src/Chapter2.hs

```diff
@@ -606,8 +625,11 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
-
+duplicate l = go [] l
+    where
+        go :: [a] -> [a] -> [a]
+        go acc [] = acc
+        go acc (x : xs) = go (acc ++ [x, x]) xs
```

Comment:
Creative idea with `go` 🎨 
 However, Adding to the end of the list `++` is not efficient 🐌 
Note that you can implement this function with the simple pattern matching as well. It requires fewer function calls and maybe (just maybe) a bit more efficient (at least by memory).